web/policies/donation_goal_policy.ex

```diff
@@ -0,0 +1,45 @@
+defmodule CodeCorps.DonationGoalPolicy do
+  import Ecto.Query
+
+  alias CodeCorps.DonationGoal
+  alias CodeCorps.OrganizationMembership
+  alias CodeCorps.Project
+  alias CodeCorps.Repo
+  alias CodeCorps.User
+  alias Ecto.Changeset
+
+  def create?(%User{admin: true}, %Changeset{}), do: true
+  def create?(%User{} = user, %Changeset{} = changeset) do
+    changeset |> get_project |> get_membership(user) |> get_role |> owner?
+  end
+
+  def update?(%User{admin: true}, %DonationGoal{}), do: true
+  def update?(%User{} = user, %DonationGoal{} = donation_goal) do
+    donation_goal |> get_project |> get_membership(user) |> get_role |> admin_or_higher?
+  end
+
+  def delete?(%User{admin: true}, %DonationGoal{}), do: true
+  def delete?(%User{} = user, %DonationGoal{} = donation_goal) do
+    donation_goal |> get_project |> get_membership(user) |> get_role |> owner?
+  end
+
+  defp get_project(%DonationGoal{project_id: project_id}), do: Project |> Repo.get(project_id)
+  defp get_project(%Changeset{changes: %{project_id: project_id}}), do: Project |> Repo.get(project_id)
+  defp get_project(_), do: nil
+
+  defp get_membership(nil, %User{}), do: nil
+  defp get_membership(%Project{organization_id: organization_id}, %User{id: user_id}) do
+    OrganizationMembership
+    |> where([m], m.member_id == ^user_id and m.organization_id == ^organization_id)
+    |> Repo.one
+  end
```

Comment:
Are we doing these in a lot of other places? Looks like could be extracted out maybe?